blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerConfigBase.scala

```diff
@@ -0,0 +1,135 @@
+package org.http4s
+package server
+package blaze
+
+import java.io.FileInputStream
+import java.security.KeyStore
+import java.security.Security
+import javax.net.ssl.{TrustManagerFactory, KeyManagerFactory, SSLContext, SSLEngine}
+import java.util.concurrent.ExecutorService
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+
+import org.http4s.blaze.channel
+import org.http4s.blaze.channel.SocketConnection
+import org.http4s.blaze.channel.nio1.NIO1SocketServerGroup
+import org.http4s.blaze.channel.nio2.NIO2SocketServerGroup
+import org.http4s.blaze.pipeline.LeafBuilder
+import org.http4s.blaze.pipeline.stages.{SSLStage, QuietTimeoutStage}
+import org.http4s.server.SSLKeyStoreSupport.StoreInfo
+import org.http4s.tls.ClientAuth
+import org.http4s.util.threads.DefaultPool
+
+import org.log4s.getLogger
+
+import scala.concurrent.duration._
+import scalaz.concurrent.{Strategy, Task}
+
+private[blaze] abstract class BlazeServerConfigBase { self: BlazeServerConfig =>
+  private[this] val logger = getLogger
+
+  def bindHttp(port: Int = 8080, host: String = "127.0.0.1"): BlazeServerConfig =
+    withSocketAddress(InetSocketAddress.createUnresolved(host, port))
+
+  def bindHttps(port: Int = 8443, host: String = "127.0.0.1", sslContext: SSLContext): BlazeServerConfig = {
+    withSocketAddress(InetSocketAddress.createUnresolved(host, port))
+    withSslContext(sslContext)
+  }
+
+  def start: Task[Server] = {
+    def aggregateService = Router(serviceMounts.reverse.map { mount =>
+      mount.prefix -> mount.service
+    }: _*)
+
+    def resolveAddress(address: InetSocketAddress) =
+      if (address.isUnresolved) new InetSocketAddress(address.getHostName, address.getPort)
+      else address
+
+    def pipelineFactory = { conn: SocketConnection =>
+      val requestAttributes =
+        (conn.local, conn.remote) match {
+          case (l: InetSocketAddress, r: InetSocketAddress) =>
+            AttributeMap(AttributeEntry(Request.Keys.ConnectionInfo, Request.Connection(l,r, true)))
+          case _ =>
+            AttributeMap.empty
+        }
+
+      def http1Stage =
+        Http1ServerStage(aggregateService, requestAttributes, serviceExecutor.getOrElse(DefaultPool), webSocketsEnabled, maxRequestLineLength, maxHeadersLength)
```

Comment:
I would still like to eliminate `DefaultPool`, which is a daemon pool, which is bad for graceful draining.